### PR TITLE
Add notes about ignoring initialization failures on contexts

### LINKED ISSUE #20130: doc/man3/EVP_DigestInit.pod: example code should reflect dynamic algo providers
[master] as of c3bd630df0c3630c66155fb8c4baf54810d24695.

The example code does
```
     md = EVP_get_digestbyname(argv[1]);
     if (md == NULL) {
         printf("Unknown message digest %s\n", argv[1]);
         exit(1);
     }

     mdctx = EVP_MD_CTX_new();
     EVP_DigestInit_ex2(mdctx, md, NULL);
     EVP_DigestUpdate(mdctx, mess1, strlen(mess1));
```

which to the best of my incomplete knowledge could crash since now "algorithm providers are dynamic" aka lazily resolved, so that EVP_DigestInit_ex2() could fail due to invalid "md"?

P.S.: if this is not true software like the postfix MTA should be notified, as they implemented according workarounds.
Thank you.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #20130

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/EVP_DigestSignInit.pod [hunk 1]
```diff
@@ -172,6 +172,10 @@ multiple times on a context and the parameters set by previous calls should be
 preserved if the I<pkey> parameter is NULL. The call then just resets the state
 of the I<ctx>.
 
+Ignoring failure of EVP_DigestSignInit() and EVP_DigestSignInit_ex() functions
```
Comment (paulidale) [inline doc/man3/EVP_DigestSignInit.pod:175]: Ignore failure returns from

File: doc/man3/EVP_DigestInit.pod [hunk 2]
```diff
@@ -651,6 +652,13 @@ are still in common use.
 For most applications the I<impl> parameter to EVP_DigestInit_ex() will be
 set to NULL to use the default digest implementation.
 
+Ignoring failure of EVP_DigestInit_ex(), EVP_DigestInit_ex2(), or
```
Comment (paulidale) [inline doc/man3/EVP_DigestInit.pod:655]: Ignoring failure returns from

File: doc/man3/EVP_DigestVerifyInit.pod [hunk 3]
```diff
@@ -161,6 +161,10 @@ multiple times on a context and the parameters set by previous calls should be
 preserved if the I<pkey> parameter is NULL. The call then just resets the state
 of the I<ctx>.
 
+Ignoring failure of EVP_DigestVerifyInit() and EVP_DigestVerifyInit_ex() functions
```
Comment (paulidale) [inline doc/man3/EVP_DigestVerifyInit.pod:164]: Same here and elsewhere

**Review by paulidale — CHANGES_REQUESTED** [review]

**Comment by t8m** [comment]
@paulidale fixup pushed

**Review by mattcaswell — APPROVED** [review]

**Review by beldmit — APPROVED** [review]
LGTM.

I'd added smth like "may cause application crash" or "may cause sending plain text data" (for ciphers), but don't think it's crucial.

**Comment by t8m** [comment]
Merged to master, 3.1, and 3.0 branches. Thank you for the reviews.

**Pull request closed without merging**